### PR TITLE
Upgrade to BuildingSync Version 2.4

### LINKED ISSUE #11: Update python library to BSync 2.4
BSyncpy needs to be regenerated for Schema Version 2.4. (Note that it also fails for 2.3.)

### PR DESCRIPTION
**Pull request by nllong**
Fixes #11

### CODE REVIEW

File: bsyncpy/bsyncpy_generator.py [hunk 1]
```diff
@@ -374,16 +373,45 @@ def do_complexType(element) -> BSElement:
...
-            choice_element = do_element(child)
-            logging.debug(f"        - choice_element: {choice_element.element_name}")
+            if child.tag == "element":
+                child_name = child.get("name")
+                child_ref = child.get("ref")
 
-            bs_element.element_children.append(
-                (child_name, choice_element.element_full_name)
-            )
+                if child_name and child_ref:
```
Comment (nllong) [inline bsyncpy/bsyncpy_generator.py:382]: This was the chunk of code that was needed to resolve the reference not found issue. In short, we were not checking for references in choice elements.

File: pyproject.toml [hunk 2]
```diff
@@ -11,7 +11,8 @@ lxml = "^4.6.3"
 [tool.poetry.dev-dependencies]
 pytest = "^6.2.5"
 pre-commit = "^2.15.0"
+virtualenv = "20.09.0"
```
Comment (nllong) [inline pyproject.toml:14]: GitHub Actions does not support 20.10.0, yet...

File: tests/test_bsyncpy.py [hunk 3]
```diff
@@ -49,13 +49,13 @@ def test_datetime():
     """
     Added for https://github.com/BuildingSync/bsyncpy/issues/6
     """
-    dt = datetime.datetime(2019, 1, 1, 0, 0, 0)
-    ts = bsync.TimeSeriesData.TimeSeries.StartTimestamp(dt)
+    dt = datetime.datetime(2019, 1, 1, 8, 24, 59)
+    ts = bsync.StartTimestamp(dt)
```
Comment (nllong) [inline tests/test_bsyncpy.py:53]: several elements were promoted to complex/simple types, so they are now at the top level.

**Review by macintoshpie — APPROVED** [review]
LGTM ⚡

**Pull request merged**